src/Illuminate/Database/DatabaseServiceProvider.php

```diff
@@ -45,6 +46,26 @@ public function register()
         $this->registerEloquentFactory();
 
         $this->registerQueueableEntityResolver();
+
+        $this->registerDBALTypes();
+    }
+
+    /**
+     * Add any custom types to the DBAL library.
+     *
+     * @return void
+     */
+    protected function registerDBALTypes()
+    {
+        if (! class_exists(Type::class)) {
+            return;
+        }
+
+        $types = config('database.dbal.types', []);
```

Comment:
Can't call the config function here.

Reply:
Why not?

Reply:
Usually we would do $this->app['config'] or something... like in other places within this file.

Reply:
Right. I get that for consistency that might be better.

```suggestion
        $types = $this->app['config']->get('database.dbal.types', []);
```

Reply:
Not just for consistency. This component has to work outside of Laravel. The only thing a service provider can do is look a the app property. It can't assume anything else.

Reply:
@GrahamCampbell a service provider is meant to only work with Laravel so I don't see that as a valid concern.